docs/source/extensions_tutorial/5_custom_api.rst

```diff
@@ -191,3 +190,93 @@ This class will have the methods ``add_container``, ``create_container``,  and `
 the property ``containers``. The ``add_container`` method will check to make sure that either an object of type
 ``MyContainer`` or a list/dict/tuple of objects of type ``MyContainer`` is passed in. ``create_container`` will
 accept the exact same arguments that the ``MyContainer`` class constructor accepts.
+
+NWBData
+--------
+
+:py:class:`~pynwb.core.NWBData` should be used to represent datasets with a *neurodata_type_def*. This section
+will discuss the available :py:class:`~pynwb.core.NWBData` subclasses for representing common dataset specifications.
+
+NWBTable
+^^^^^^^^^
+
+If your specification extension contains a table definition i.e. a dataset with a compound data type, you should use
+the :py:class:`~pynwb.core.NWBTable` class to represent this specification. Since :py:class:`~pynwb.core.NWBTable`
+subclasses :py:class:`~pynwb.core.NWBData`, you can still use ``__nwbfields__``. In addition, you can use the
+``__columns__`` class property to specify the columns of the table. ``__columns__`` should be a list or a tuple of
+:py:func:`~hdmf.utils.docval`-like dictionaries.
+
+The following example demonstrates how to define a table with the columns ``foo`` and ``bar`` that are of type
+str and int, respectively. We also register the class as the representation of the *neurodata_type* "MyTable"
+from the *namespace* "my_ns".
+
+.. code-block:: python
+
+    from pynwb import register_class
+    from pynwb.core import NWBTable
+
+
+    @register_class('MyTable', 'my_ns')
+    class MyTable(NWBTable):
+
+        __columns__ = [
+            {'name': 'foo', 'type': str, 'doc': 'the foo column'},
+            {'name': 'bar', 'type': int, 'doc': 'the bar column'},
+        ]
+
+        ...
+
+NWBTableRegion
+^^^^^^^^^^^^^^
+
+:py:class:`~pynwb.core.NWBTableRegion` should be used to represent datasets that store a region reference. The constructor
+for :py:class:`~pynwb.core.NWBTableRegion`. When subclassing this class, make sure you provide a way to pass in the required
+arguments for the :py:class:`~pynwb.core.NWBTableRegion` constructor--the *name* of the dataset, the *table* that the region
+applies to, and the *region* itself.
```

Comment:
Do we use this and is it well tested? If not, it might be best not to recommend it / document it.